Datadog.Serverless.Compat/DatadogServerlessCompat.cs

```diff
@@ -0,0 +1,184 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using Microsoft.Extensions.Logging;
+
+namespace Datadog.Serverless.Compat
+{
+    public enum CloudEnvironment
+    {
+        AzureFunction,
+        AzureSpringApp,
+        GoogleCloudRunFunction1stGen,
+        Unknown
+    }
+
+    public static class DatadogServerlessCompat
```

Comment:
If this is really public API (the user, not the tracer, will call the `Start()` method), we should probably reconsider this class name and the namespace.

Right now the full name is `Datadog.Serverless.Compat.DatadogServerlessCompat` which doesn't really follow .NET conventions.

Maybe something like:
```
namespace: "Datadog.Serverless"
class: "CompatibilityLayer" or "Agent"
```

Reply:
Updated here: https://github.com/DataDog/datadog-serverless-compat-dotnet/commit/95919ee2b873e0b0be0e67bad95a53ee04a1184a

What do you think about `Datadog.Serverless` as the namespace/package name and `Compat` as the the class name? `Datadog.Serverless.Compat` would be consistent with the other runtime packages
- https://pypi.org/project/datadog-serverless-compat/
- https://www.npmjs.com/package/@datadog/serverless-compat
- https://mvnrepository.com/artifact/com.datadoghq/dd-serverless-compat-java-agent

Reply:
I think it's more important to follow each ecosystem's naming conventions so things feel idiomatic rather than to have consistency across languages 😅

But if we're married to `Datadog.Serverless.Compat`, then I think we should use:
- nuget package id: `Datadog.Serverless.Compat`
- project name: `Datadog.Serverless.Compat.csproj`
- assembly name: `Datadog.Serverless.Compat.dll`
- C# namespace: `Datadog.Serverless`
- C# class name: `Compat`